app/models/performance_statistics.rb

```diff
@@ -0,0 +1,33 @@
+class PerformanceStatistics
+  QUERY = "
+  WITH raw_data AS (
+      SELECT
+          SUM(CASE WHEN f.id IS NOT NULL THEN 1 ELSE 0 END) candidate_forms,
```

Comment:
I think you can use `COUNT FILTER` for these kind of stats, e.g.

```
COUNT(c.id) FILTER (WHERE f.id IS NOT NULL) AS candidate_forms,
```

Reply:
Maybe with `COUNT(DISTINCT c.id)` you could do without the common table expression and do the query in 'one go'. (I'm not clear about under what circumstances you could have a candidate with more than one application form).

Reply:
Thanks @stevehook, I've incorporated `COUNT FILTER` into the query.

I've also added some model specs which should make it easier to refactor stuff to make the query simpler, although I probably would hold off on that in this PR (my SQL is very rusty at this stage).